### PR TITLE
Appointment changes

### PR DESCRIPTION
**Pull request by rgforsyth**
## What does this pull request do?

Moves Appointment into its own table.
Rename action_plan_appointment to session_delivery_appointment

## What is the intent behind these changes?

Make Appointment generic so that it can be used for multiple appointment types.

### CODE REVIEW

**Comment by rgforsyth** [comment]
Still need to look at fixing the sql report, but raising pr for early sight.

File: helm_deploy/hmpps-interventions-service/reports/crs_performance_report.sql [hunk 1]
```diff
@@ -15,9 +15,10 @@ COPY (
     'coming-later'          AS date_of_first_action_plan_approval,
     r.assigned_to_id        AS caseworker_id,
     (
-      select min(app.appointment_time)
-      from action_plan_appointment app
-      where app.action_plan_id = ap.id and attended in ('YES', 'LATE')
+        select min(app.appointment_time)
```
Comment (rgforsyth) [inline helm_deploy/hmpps-interventions-service/reports/crs_performance_report.sql:18]: Have not been able to test these changes locally

File: src/main/resources/db/migration/V1_54__appointment.sql [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    attendance_behaviour text,
+    notifyppof_attendance_behaviour boolean,
+    attendance_behaviour_submitted_at timestamp with time zone,
+
+    session_feedback_submitted_at timestamp with time zone,
+    delius_appointment_id bigint,
+
+    constraint pk_appt_id primary key (id),
+    constraint fk_appt_to_created_by foreign key (created_by_id) references auth_user
+);
```
Comment (tomsmyers) [inline src/main/resources/db/migration/V1_54__appointment.sql:23]: can we copy over existing appointments into the new table before dropping the old columns?

**Comment by tomsmyers** [comment]
please can you include the jira ticket number in the PR title, so it links up with jira ?

File: helm_deploy/hmpps-interventions-service/reports/crs_performance_report.sql [hunk 3]
```diff
@@ -15,9 +15,10 @@ COPY (
...
+        from session_delivery_appointment ses
+        join appointment app on ses.appointment_id = app.id
+        where ses.action_plan_id = ap.id and app.attended in ('YES', 'LATE')
```
Comment (tomsmyers) [inline helm_deploy/hmpps-interventions-service/reports/crs_performance_report.sql:21]: typo here `ap.id`

Comment (tomsmyers) [inline helm_deploy/hmpps-interventions-service/reports/crs_performance_report.sql:21]: i know it's a copy paste from the original, but where is `ap` defined?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/dto/ActionPlanAppointmentsDTO.kt [hunk 4]
```diff
@@ -41,25 +41,25 @@ data class ActionPlanAppointmentDTO(
...
+          actionPlanSession.appointment.additionalAttendanceInformation,
+          actionPlanSession.appointment.attendanceBehaviour,
+          actionPlanSession.appointment.notifyPPOfAttendanceBehaviour,
+          actionPlanSession.appointment.attendanceSubmittedAt != null,
         ),
       )
     }
-    fun from(appointments: List<ActionPlanAppointment>): List<ActionPlanAppointmentDTO> {
-      return appointments.map { from(it) }
+    fun from(actionPlanSession: List<ActionPlanSession>): List<ActionPlanSessionDTO> {
```
Comment (tomsmyers) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/dto/ActionPlanAppointmentsDTO.kt:61]: TINY nit: missing an 's' from 'actionPlanSessions'

**Pull request closed without merging**